app/src/main/java/com/hedvig/app/feature/home/model/HomeItemsBuilder.kt

```diff
@@ -0,0 +1,163 @@
+package com.hedvig.app.feature.home.model
+
+import androidx.compose.ui.unit.dp
+import arrow.core.NonEmptyList
+import com.hedvig.android.owldroid.graphql.HomeQuery
+import com.hedvig.android.owldroid.type.PayinMethodStatus
+import com.hedvig.app.R
+import com.hedvig.app.feature.claims.ui.commonclaim.CommonClaimsData
+import com.hedvig.app.feature.claims.ui.commonclaim.EmergencyData
+import com.hedvig.app.feature.home.ui.claimstatus.data.ClaimStatusCardUiState
+import com.hedvig.app.util.featureflags.Feature
+import com.hedvig.app.util.featureflags.FeatureManager
+
+@OptIn(ExperimentalStdlibApi::class)
+class HomeItemsBuilder(
+    private val featureManager: FeatureManager
+) {
+
+    fun buildItems(
+        homeData: HomeQuery.Data
+    ): List<HomeModel> = when {
+        homeData.isActive() -> buildActiveItems(homeData)
+        (homeData.isPending() || homeData.isActiveInFuture()) && homeData.isSwitching() -> buildSwitchingItems(homeData)
```

Comment:
Super small nit, but could we flip the things around the && to put the isSwitching at the front to indicate that this is predominantly why this case exists? At first I read the isPending and thought  how can we have this here and in the line below, but it's all because the isSwitching is the most important thing in this expression.
Again you don't need to do this, but it got me confused initially. Maybe it wouldn't inside the IDE where I could've seen the brackets easier

Reply:
Fixed!